packages/@aws-cdk/aws-cloudwatch/lib/anomaly-detection-alarm.ts

```diff
@@ -0,0 +1,370 @@
+import { dispatchMetric } from './private/metric-util';
+import { MetricSet } from './private/rendering';
+
+/**
+ * Options for an AnomolyDetectionAlarm
+ */
+export interface AnomalyDetectionAlarmOptions {
+
+  /**
+   * Name of the alarm
+   *
+   * @default - Automatically generated name
+   */
+  readonly alarmName?: string;
+
+  /**
+   * Description for the alarm
+   *
+   * @default - No description
+   */
+  readonly alarmDescription?: string;
+
+  /**
+   * Comparison to use to check if metric is breaching
+   *
+   * @default - AnomalyDetectionComparisonOperator.LESS_THAN_LOWER_OR_GREATER_THAN_UPPER_THRESHOLD
+   */
+  readonly comparisonOperator?: AnomalyDetectionComparisonOperator;
+
+  /**
+   * The number of standard deviations used when generating the anomaly detection band.
+   *
+   * Higher number means thicker band, lower number means thinner band.
+   */
+  readonly threshold: number;
+
+  /**
+   * The number of periods over which data is compared to the specified threshold.
+   */
+  readonly evaluationPeriods: number;
+
+  /**
+   * Sets how this alarm is to handle missing data points.
+   *
+   * @default - TreatMissingData.MISSING
+   */
+  readonly treatMissingData?: TreatMissingData;
+
+  /**
+   * Whether the actions for this alarm are enabled
+   *
+   * @default - true
+   */
+  readonly actionsEnabled?: boolean;
+
+  /**
+   * The number of datapoints that must be breaching to trigger the alarm. This is used only if you are setting an "M
+   * out of N" alarm. In that case, this value is the M. For more information, see Evaluating an Alarm in the Amazon
+   * CloudWatch User Guide.
+   *
+   * @default - same as evaluationPeriods
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarm-evaluation
+   */
+  readonly datapointsToAlarm?: number;
+}
+
+/**
+ * Properties for AnomolyDetectionAlarms
+ */
+export interface AnomalyDetectionAlarmProps extends AnomalyDetectionAlarmOptions {
+  /**
+   * The metric to add the alarm on
+   *
+   * Metric objects can be obtained from most resources, or you can construct
+   * custom Metric objects by instantiating one.
+   */
+  readonly metric: IMetric;
+}
+
+/**
+ * Comparison operator for evaluating anomaly detection alarms
+ */
+export enum AnomalyDetectionComparisonOperator {
+  /**
+   * Specified statistic is lower than or greater than the anomaly model band.
+   */
+  LESS_THAN_LOWER_OR_GREATER_THAN_UPPER_THRESHOLD = 'LessThanLowerOrGreaterThanUpperThreshold',
+
+  /**
+   * Specified statistic is greater than the anomaly model band.
+   */
+  GREATER_THAN_UPPER_THRESHOLD = 'GreaterThanUpperThreshold',
+
+  /**
+   * Specified statistic is lower than the anomaly model band.
+   */
+  LESS_THAN_LOWER_THRESHOLD = 'LessThanLowerThreshold',
+}
+
+/**
+ * An anomaly detection alarm on a CloudWatch metric.
+ *
+ * @resource AWS::CloudWatch::Alarm
+ */
+export class AnomalyDetectionAlarm extends AlarmBase {
```

Comment:
It looks like a lot of this is duplicating what's in the `Alarm` class. We should abstract the duplications out of both into a class that these can extend.